man/journalctl.xml

```diff
@@ -360,6 +360,18 @@
       </varlistentry>
 
       <varlistentry>
+        <term><option>--no-hostname</option></term>
+
+        <listitem><para>Don't show the hostname field of log messages originating from the local host. Specifically,
+        this suppresses output of the hostname column for all log messages generated with the same machine ID as the
+        local system (see
+        <citerefentry><refentrytitle>machine-id</refentrytitle><manvolnum>5</manvolnum></citerefentry> for
+        details). Note that frequently early boot messages are generated without the local hostname or machine ID known
+        or defined, and are hence unaffected by this switch. This switch only has an effect on the
+        <option>short</option> family of output modes (see above).</para></listitem>
+      </varlistentry>
```

Comment:
I think this is unnecessarily restrictive. The common use case is looking at the logs from the local machine, where you **know** that all the logs are local, even if the hostname/machine-id don't match, and having the hostname displayed for some of the messages is still useless.